administrator/components/com_userlogs/models/userlogs.php

```diff
@@ -16,131 +16,148 @@
+
+		// Apply ordering
+		if (!empty($fullorderCol))
+		{
+			$query->order($db->escape($fullorderCol));
+		}
+		// Get filter by user
+		$user = $this->getState('filter.user');
+
+		// Apply filter by user
+		if (!empty($user))
+		{
+			$query->where($db->quoteName('a.user_id') . ' = ' . (int) $user);
+		}
+
+		// Get filter by extension
+		$extension = $this->getState('filter.extension');
+
+		// Apply filter by extension
+		if (!empty($extension))
+		{
+			$query->where($db->quoteName('a.extension') . ' = ' . $db->quote($extension));
+		}
+
+		// Get filter by date range
+		$dateRange = $this->getState('filter.dateRange');
+
+		// Apply filter by date range
+		if (!empty($dateRange))
+		{
+			$date = $this->buildDateRange($dateRange);
+
+			// If the chosen range is not more than a year ago
+			if ($date['dNow'] != false)
+			{
+				$query->where(
+					$db->qn('a.log_date') . ' >= ' . $db->quote($date['dStart']->format('Y-m-d H:i:s')) .
+					' AND ' . $db->qn('a.log_date') . ' <= ' . $db->quote($date['dNow']->format('Y-m-d H:i:s'))
+				);
+			}
+		}
+
+		// Filter the items over the search string if set.
+		$search = $this->getState('filter.search');
+
+		if (!empty($search))
+		{
+			$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
 			$query->where('(a.message LIKE ' . $search . ')');
-        }
-
-        return $query;
-    }
-
-    protected function checkIn()
-    {
-        $plugin = JPluginHelper::getPlugin('system', 'userlogs');
-        if(!empty($plugin))
-        {
-            $pluginParams = new JRegistry($plugin->params);
-            $daysToDeleteAfter = (int)$pluginParams->get('logDeletePeriod');
-
-            if ($daysToDeleteAfter > 0) {
-                $db = JFactory::getDbo();
-                $query = $db->getQuery(true);
-                $conditions = array( $db->quoteName('log_date') .' < DATE_SUB(NOW(), INTERVAL '. $daysToDeleteAfter .' DAY)');
-                $query->delete($db->quoteName('#__user_logs'));
-                $query->where($conditions);
-                $db->setQuery($query);
-                $result = $db->execute();
-            }
-        }
-    }
-
-    /**
+		}
+
+		return $query;
+	}
+
+	/**
+	 * Check for old logs that needs to be deleted_comment
+	 *
+	 * @return void
+	 *
+	 * @since 3.6
+	 */
+	protected function checkIn()
+	{
+		$plugin = JPluginHelper::getPlugin('system', 'userlogs');
+
+		if (!empty($plugin))
+		{
+			$pluginParams = new JRegistry($plugin->params);
+			$daysToDeleteAfter = (int) $pluginParams->get('logDeletePeriod');
+
+			if ($daysToDeleteAfter > 0)
+			{
+				$db = JFactory::getDbo();
+				$query = $db->getQuery(true);
+				$conditions = array( $db->quoteName('log_date') . ' < DATE_SUB(NOW(), INTERVAL ' . $daysToDeleteAfter . ' DAY)');
+				$query->delete($db->quoteName('#__user_logs'));
+				$query->where($conditions);
+				$db->setQuery($query);
+				$result = $db->execute();
```

Comment:
I mean that line: 

$result = $db->execute();

You don't do anything with the result :-)